### PR TITLE
Fix login

### PR DESCRIPTION
**Pull request by kabiroberai**
## Notion ticket link
N/A

## Implementation description
* This PR fixes login, which was failing because AuthDTO didn't know about UserDTO's `branch` field.

## Steps to test
1. Try creating an account and logging in (`/login`) before and after this PR.
2. Observe that there's a 500 error prior to applying this patch.

## What should reviewers focus on?
* test_auth_service: This is my first time writing a pytest-based unit test so feedback would be great!

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/tests/functional/test_auth_service.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        "auth_id": "A",
+        "first_name": "Jane",
+        "last_name": "Doe",
+        "role": "Admin",
+    },
+)
+
+
+def insert_users():
```
Comment (kabiroberai) [inline backend/python/tests/functional/test_auth_service.py:59]: this is taken from test_user_service.py; I wonder whether we should unify the two methods or whether that's taking DRY too far.

File: backend/python/tests/functional/test_auth_service.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    user_instances = [User(**data) for data in TEST_USERS]
+    db.session.bulk_save_objects(user_instances)
+    db.session.commit()
+
+
+def test_login(auth_service):
+    insert_users()
+    auth_service.generate_token("test@test.com", "password")
```
Comment (kabiroberai) [inline backend/python/tests/functional/test_auth_service.py:67]: No asserts here but login failures throw an exception, and this was failing as expected before applying the auth_dto patch.

**Review by jennchenn — APPROVED** [review]
lgtm thanks for catching this

let's leave the tests as is for now, they can be refactored in the future if necessary

File: backend/python/tests/functional/test_auth_service.py (hunk 2 above)
Comment (jennchenn) [reply backend/python/tests/functional/test_auth_service.py:67]: so is the test here just that these operations do not throw an exception?

Comment (kabiroberai) [reply backend/python/tests/functional/test_auth_service.py:67]: Pretty much, yep

**Pull request merged**